googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,222 @@
+        builder.set(fieldName, Integer.parseInt(map.get(fieldName)));
+      } else {
+        builder.set(fieldName, map.get(fieldName));
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * {@link PluginConfig} class for {@link BigQuerySource}
+   */
+  public static class BQSourceConfig extends PluginConfig {
+    private static final String IMPORT_QUERY = "importQuery";
+    private static final String PROJECT_ID = "projectId";
+    private static final String INPUT_TABLE_ID = "InputTableId";
+    private static final String JSON_FILE_PATH = "jsonFilePath";
+    private static final String TEMP_BUCKET = "tempBuketPath";
+    private static final String OUTPUT_SCHEMA = "outputSchema";
+    private static final String PROJECTID_DESC = "The ID of the project in Google Cloud";
+    private static final String TEMP_BUCKET_DESC = "the tempory google cloud storage directory to store the " +
+                                                   "intermediate result. Example: gs://bucketname/directoryname, " +
+                                                   "the directory should not be existed. User should delete this " +
+                                                   "directory afterward manually to avoid extra google storage charge.";
+    private static final String IMPORT_QUERY_DESC = "The SELECT query to use to import data from the specified table." +
+                                                    " example: SELECT TOP(corpus, 10) as title, COUNT(*) as " +
+                                                    "unique_words FROM [publicdata:samples.shakespeare], where " +
+                                                    "publicdata is the project name, smalples is the dataset name, " +
+                                                    "shakespare is the table name. This is optional, if empty, just " +
+                                                    "read the inputTable configured";
+    private static final String JSON_KEYFILE_DESC = "the credential json key file path";
+    private static final String INPUT_TABLE_DESC = "The BigQuery table to read from, in the form " +
+                                                   "[optional projectId]:[datasetId].[tableId]. Example: " +
+                                                   "publicdata:samples.shakespeare. Note that if the import query is " +
+                                                   "specified, this table should be a empty table with the " +
+                                                   "query result schema. User need to first create such a table";
+    private static final String OUTPUTSCHEMA_DESC = "Comma separated mapping of column names in the output schema to " +
+                                                   "the data types; for example: 'A:string,B:int'";
+    @Name(Constants.Reference.REFERENCE_NAME)
+    @Description(Constants.Reference.REFERENCE_NAME_DESCRIPTION)
+    public String referenceName;
```

Comment:
@yug095 I think you should make this config class extend `ReferencePluginConfig` instead as it already has this property.